### PR TITLE
Modify run-jenkins.sh to adapt wasm build for the OSX CI lane

### PR DESCRIPTION
**Pull request by steveisok**
Relates to https://github.com/mono/mono/issues/16858

### CODE REVIEW

**Comment by steveisok** [comment]
@monojenkins build OS X WebAssembly

**Comment by steveisok** [comment]
@monojenkins build OS X WebAssembly

File: scripts/ci/run-jenkins.sh [hunk 1]
```diff
@@ -325,16 +329,24 @@ if [[ ${CI_TAGS} == *'webassembly'* ]] || [[ ${CI_TAGS} == *'wasm'* ]];
...
-	   ${TESTCMD} --label=archive   --timeout=180m --fatal $gnumake -j ${CI_CPU_COUNT} --output-sync=recurse --trace -C sdks/builds archive-wasm   NINJA=
-
+        export aot_test_suites="System.Core"
+        export mixed_test_suites="System.Core"
+        export xunit_test_suites="System.Core corlib"
+
+        ${TESTCMD} --label=provision --timeout=20m --fatal $gnumake --output-sync=recurse --trace -C sdks/builds provision-wasm
+
+        if [[ ${CI_TAGS} == *'osx-amd64'* ]]; then
+            ${TESTCMD} --label=build-runtime     --timeout=180m --fatal $gnumake -j ${CI_CPU_COUNT} --output-sync=recurse --trace -C sdks/wasm runtime
```
Comment (vargaz) [inline scripts/ci/run-jenkins.sh]: Why doesn't the other version work on osx ?

**Comment by steveisok** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by jaykrell** [comment]
Having a non-Linux WebAssembly lane is nice.
Having a Windows WebAssembly lane would be nice too. :)

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by MaximLipnin** [comment]
Finally `OS X WebAssembly` lane is green.

**Comment by MaximLipnin** [comment]
@monojenkins build failed

File: sdks/wasm/Makefile [hunk 2]
```diff
@@ -302,7 +306,11 @@ define XunitTestTemplate
 
 bin/$(2)/mono.js: packager.exe $(WASM_FRAMEWORK)/.stamp-framework xunit-runner.exe runtime-tests.js xunit-exclusions.rsp
 	mono packager.exe --debugrt --search-path=$(XUNIT_DIR) --search-path=$(WASM_BCL_DIR)/tests --template=runtime-tests.js --appdir=bin/$(2) --builddir=obj/$(2) --asset=xunit-exclusions.rsp xunit-runner.exe $(XUNIT_DIR)/xunit.execution.dotnet.dll $(WASM_BCL_DIR)/tests/$(3) $(WASM_BCL_DIR)/tests/Xunit.NetCore.Extensions.dll
+ifeq ($(UNAME),Darwin)
+	ninja -v -C obj/$(2) -w dupbuild=warn
+else
```
Comment (vargaz) [inline sdks/wasm/Makefile]: Ditto.

File: sdks/wasm/Makefile [hunk 3]
```diff
@@ -280,7 +280,11 @@ TEST_ASSEMBLIES = $(WASM_BCL_DIR)/nunitlite.dll $(WASM_BCL_DIR)/tests/wasm_corli
 
 build-test-suite: packager.exe $(WASM_FRAMEWORK)/.stamp-framework binding_tests.dll mini_tests.dll main.exe runtime-tests.js
 	mono packager.exe --emscripten-sdkdir=$(EMSCRIPTEN_SDK_DIR) --template=runtime-tests.js --appdir=bin/test-suite --builddir=obj/test-suite --asset=tests/runtime/index.html binding_tests.dll mini_tests.dll main.exe $(TEST_ASSEMBLIES) --linker
+ifeq ($(UNAME),Darwin)
+	ninja -v -C obj/test-suite -w dupbuild=warn
+else
```
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply sdks/wasm/Makefile]: Was it fixed?

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

File: sdks/wasm/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaximLipnin) [reply sdks/wasm/Makefile]: Should be fixed in https://github.com/mono/mono/pull/17596/commits/ee008feaa0ea1d760d7d6911adcb6cd6a2c128cb

File: scripts/ci/run-jenkins.sh (hunk 1 above)
Comment (akoeplinger) [reply scripts/ci/run-jenkins.sh]: @MaximLipnin we should check whether this is still necessary with the latest PR job config.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by MaximLipnin** [comment]
@monojenkins build OS X WebAssembly

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by MaximLipnin** [comment]
the most green I've ever seen

**Pull request merged**